### PR TITLE
feat: add COE static constructors

### PR DESCRIPTION
**Pull request by vishwa2710**
- Implemented methods to compute Sun-synchronous inclination and RAAN from LTAN in the COE class.
- Added static methods for constructing Sun-synchronous, Geosynchronous COEs, Circular, Equatorial, CircularEquatorial COEs

## Summary by CodeRabbit

* **New Features**
  * Added high‑level orbit constructors: sun‑synchronous, stationary (geo‑synchronous renamed), circular, equatorial, plus utilities to compute sun‑synchronous inclination and RAAN from local time.

* **Documentation**
  * Clarified docstrings; sun_synchronous marks argument_of_latitude as optional with a default.

* **Refactor**
  * Replaced manual element assembly with COE factory constructors; public API renamed geo_synchronous → stationary.

* **Tests**
  * Expanded Python and C++ tests covering new constructors, computations, defaults and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/COE.hpp [hunk 1]
```diff
@@ -454,6 +454,88 @@ class COE
...
+    /// @param anInclination An inclination
+    /// @param aLongitude A longitude above the surface
+    /// @param aCelestialObjectSPtr A shared pointer to a central celestial body
+    /// @return COE
+    static COE GeoSynchronous(
+        const Instant& anEpoch,
+        const Angle& anInclination,
+        const Angle& aLongitude,
+        const Shared<const Celestial>& aCelestialObjectSPtr
+    );
```
Comment (phc1990) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/COE.hpp]: Would it make sense to default inclination like it's been done for other params in the other constructors?

```suggestion
    static COE GeoSynchronous(
        const Instant& anEpoch,
        const Angle& aLongitude,
        const Shared<const Celestial>& aCelestialObjectSPtr,
        const Angle& anInclination = Angle::Zero()
    );
```

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/COE.cpp [hunk 2]
```diff
@@ -1099,6 +1112,319 @@ Time COE::ComputeLTDN(const Angle& raan, const Instant& anInstant, const Sun& aS
...
+            return Angle::Radians(std::acos(cos_i));
+        }
+
+        cos_i = cos_i - f / df;
+    }
+
+    throw ostk::core::error::RuntimeError("Newton-Raphson did not converge.");
+}
+
+Angle COE::ComputeRaanFromLTAN(const Time& aLocalTimeAtAscendingNode, const Instant& anEpoch, const Sun& aSun)
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/COE.cpp]: Yeah that's fair, I added a celestial object as an input and raising an error if it's not an earth. Worth mentioning all the other ComputeLTAN etc. methods are earth specific right now.
